ledger-core/virtual/integration/method_test.go

```diff
@@ -348,867 +390,540 @@ func TestVirtual_CallMethodAfterPulseChange(t *testing.T) {
+	t.Log("C5091")
+	table := []struct {
+		name   string
+		stateB contract.MethodIsolation
+	}{
+		{
+			name:   "A.New calls ordered B.Foo",
+			stateB: tolerableFlags(),
+		}, {
+			name:   "A.New calls unordered B.Foo",
+			stateB: intolerableFlags(),
+		},
 	}
-	msg := server.WrapPayload(&pl).Finalize()
-	server.SendMessage(ctx, msg)
-
-	// wait for all calls and SMs
-	testutils.WaitSignalsTimed(t, 10*time.Second, executeDone)
-	testutils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
-
-	require.Equal(t, 1, typedChecker.VCallRequest.Count())
-	require.Equal(t, 2, typedChecker.VCallResult.Count())
-}
+	for _, test := range table {
+		t.Run(test.name, func(t *testing.T) {
 
-// unordered A.Foo sends ordered outgoing and receives error
-func TestVirtual_CallContractFromContract_InterferenceViolation(t *testing.T) {
-	t.Log("C4980")
+			mc := minimock.NewController(t)
 
-	mc := minimock.NewController(t)
+			server, ctx := utils.NewUninitializedServer(nil, t)
+			defer server.Stop()
 
-	server, ctx := utils.NewUninitializedServer(nil, t)
-	defer server.Stop()
+			logger := inslogger.FromContext(ctx)
 
-	logger := inslogger.FromContext(ctx)
+			executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 2)
 
-	executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
+			runnerMock := logicless.NewServiceMock(ctx, mc, func(execution execution.Context) string {
+				return execution.Request.CallSiteMethod
+			})
+			server.ReplaceRunner(runnerMock)
+			server.Init(ctx)
+			server.IncrementPulseAndWaitIdle(ctx)
 
-	runnerMock := logicless.NewServiceMock(ctx, mc, func(execution execution.Context) string {
-		return execution.Request.CallSiteMethod
-	})
-	server.ReplaceRunner(runnerMock)
-	server.Init(ctx)
-	server.IncrementPulseAndWaitIdle(ctx)
-	var (
-		class = gen.UniqueGlobalRef()
+			typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
 
-		objectAGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+			var (
+				isolation = contract.ConstructorIsolation()
+				callFlags = payload.BuildCallFlags(isolation.Interference, isolation.State)
 
-		flags = contract.MethodIsolation{
-			Interference: contract.CallIntolerable,
-			State:        contract.CallDirty,
-		}
-	)
+				classA        = gen.UniqueGlobalRef()
+				outgoingA     = server.RandomLocalWithPulse()
+				objectAGlobal = reference.NewSelf(outgoingA)
 
-	Method_PrepareObject(ctx, server, payload.Ready, objectAGlobal)
+				classB        = gen.UniqueGlobalRef()
+				objectBGlobal = reference.NewSelf(server.RandomLocalWithPulse())
 
-	outgoingCallRef := gen.UniqueGlobalRef()
+				outgoingCallRef = gen.UniqueGlobalRef()
+			)
 
-	expectedError, err := foundation.MarshalMethodErrorResult(throw.E("interference violation: ordered call from unordered call"))
-	require.NoError(t, err)
+			Method_PrepareObject(ctx, server, payload.Ready, objectBGlobal)
 
-	expectedResult := []byte("finish A.Foo")
+			// add ExecutionMocks to runnerMock
```

Comment:
It will be nice to use these `{}` structures in other table tests, that you have written.

Reply:
done